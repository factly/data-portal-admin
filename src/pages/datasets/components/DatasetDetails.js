import React from 'react';
import { useHistory } from 'react-router-dom';
import { Descriptions, Card } from 'antd';
import moment from 'moment';
import { EditOutlined } from '@ant-design/icons';

const DatasetDetails = ({ dataset }) => {
  const history = useHistory();
  return (
    <Card>
      <Descriptions
        title={dataset.title}
        column={1}
        extra={
          <EditOutlined
            key="edit"
            style={{ fontSize: '150%' }}
            onClick={() => history.push(`/datasets/${dataset.id}/edit`)}
          />
        }
        bordered
      >
        <Descriptions.Item label="Organisation">{dataset.organisation}</Descriptions.Item>
        <Descriptions.Item label="Sectors">{dataset.sectors}</Descriptions.Item>
        <Descriptions.Item label="Contact email">{dataset.contact_email}</Descriptions.Item>
        <Descriptions.Item label="Contact name">{dataset.contact_name}</Descriptions.Item>
        <Descriptions.Item label="Data standard">{dataset.data_standard}</Descriptions.Item>
        <Descriptions.Item label="Description">
          <div
            style={{ whiteSpace: 'pre-wrap' }}
            dangerouslySetInnerHTML={{ __html: dataset.description }}
          />
        </Descriptions.Item>
        <Descriptions.Item label="Units">
          <div
            style={{ whiteSpace: 'pre-wrap' }}
            dangerouslySetInnerHTML={{ __html: dataset.units }}
          />
        </Descriptions.Item>
        <Descriptions.Item label="Frequency">{dataset.frequency}</Descriptions.Item>
        <Descriptions.Item label="Granularity">{dataset.granularity}</Descriptions.Item>
        <Descriptions.Item label="License">{dataset.license}</Descriptions.Item>
        <Descriptions.Item label="Related articles">{dataset.related_articles}</Descriptions.Item>
        <Descriptions.Item label="Source">{dataset.source}</Descriptions.Item>
        <Descriptions.Item label="Source Link">{dataset.source_link}</Descriptions.Item>
        <Descriptions.Item label="Archive Link">{dataset.archive_link}</Descriptions.Item>
        <Descriptions.Item label="Temporal coverage">{dataset.temporal_coverage}</Descriptions.Item>
        <Descriptions.Item label="Time saved">{dataset.time_saved}</Descriptions.Item>
        <Descriptions.Item label="Profiling URL">
          <a>{dataset.profiling_url}</a>
        </Descriptions.Item>
        <Descriptions.Item label="Is Public">{dataset.is_public ? 'yes' : 'no'}</Descriptions.Item>
        <Descriptions.Item label="Created At">
          {moment(dataset.created_at).format('Do MMMM YYYY')}
        </Descriptions.Item>
        <Descriptions.Item label="Last Updated At">
          {moment(dataset.updated_at).format('Do MMMM YYYY')}
        </Descriptions.Item>
        <Descriptions.Item label="Next Update">
          {dataset.next_update ? moment(dataset.next_update).format('Do MMMM YYYY') : null}
        </Descriptions.Item>
      </Descriptions>
    </Card>
  );
};

export default DatasetDetails;
