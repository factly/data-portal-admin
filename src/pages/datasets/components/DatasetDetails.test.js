import React from 'react';
import { shallow } from 'enzyme';

import DatasetDetails from './DatasetDetails';

describe('DatasetDetails component', () => {
  const dataset = {
    id: 1,
    title: 'title',
    contact_email: 'contact_email',
    contact_name: 'contact_name',
    data_standard: 'data_standard',
    description: 'description',
    frequency: 'frequency',
    granularity: 'granularity',
    license: 'license',
    related_articles: 'related_articles',
    source: 'source',
    temporal_coverage: 'temporal_coverage',
    time_saved: 'time_saved',
    created_at: '2020-12-12',
  };
  const dataset2 = {
    id: 2,
    title: 'title 2',
    contact_email: 'contact_email',
    contact_name: 'contact_name',
    data_standard: 'data_standard',
    description: 'description',
    frequency: 'frequency',
    granularity: 'granularity',
    license: 'license',
    related_articles: 'related_articles',
    source: 'source',
    temporal_coverage: 'temporal_coverage',
    time_saved: 'time_saved',
    is_public: 'yes',
    created_at: '2020-12-12',
  };

  it('should render the component', () => {
    const component = shallow(<DatasetDetails dataset={dataset} />);
    expect(component).toMatchSnapshot();
  });
  it('should render the component with public dataset', () => {
    const component = shallow(<DatasetDetails dataset={dataset2} />);
    expect(component).toMatchSnapshot();
  });
});
