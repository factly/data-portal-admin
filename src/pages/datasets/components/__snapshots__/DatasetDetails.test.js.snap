// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetDetails component should render the component 1`] = `
<Card>
  <Descriptions
    column={1}
    title="title"
  >
    <DescriptionsItem
      label="Contact email"
    >
      contact_email
    </DescriptionsItem>
    <DescriptionsItem
      label="Contact name"
    >
      contact_name
    </DescriptionsItem>
    <DescriptionsItem
      label="Data standard"
    >
      data_standard
    </DescriptionsItem>
    <DescriptionsItem
      label="Description"
    >
      description
    </DescriptionsItem>
    <DescriptionsItem
      label="Frequency"
    >
      frequency
    </DescriptionsItem>
    <DescriptionsItem
      label="Granularity"
    >
      granularity
    </DescriptionsItem>
    <DescriptionsItem
      label="License"
    >
      license
    </DescriptionsItem>
    <DescriptionsItem
      label="Related articles"
    >
      related_articles
    </DescriptionsItem>
    <DescriptionsItem
      label="Source"
    >
      source
    </DescriptionsItem>
    <DescriptionsItem
      label="Temporal coverage"
    >
      temporal_coverage
    </DescriptionsItem>
    <DescriptionsItem
      label="Time saved"
    >
      time_saved
    </DescriptionsItem>
    <DescriptionsItem
      label="Created At"
    >
      in 3 months
    </DescriptionsItem>
  </Descriptions>
</Card>
`;
